### ISSUE TITLE
Julia (and other `juliaup` commands) no longer run with latest version of `juliaup` on intel mac

### ISSUE THREAD

**Issue by jbisits** [JuliaLang/juliaup#710]
I have been using `juliaup` for around a year now and it is great, so thanks for all the hard work on it! After the latest version of `juliaup` was installed I can no longer run julia. Instead I first get this error message

```terminal
julia
Error: Failed to run version db update

Caused by:
    0: Failed to start juliaup for version db update.
    1: Permission denied (os error 13)
```

then the julia repl temporarily appears but after pressing any key julia exits with the long error message pasted at the bottom of this issue. I also cannot run any `juliaup` commands

```terminal
juliaup list
zsh: permission denied: juliaup
```

This could be related (or a duplicate) to #624 but I have checked the read and write access and everything looks to be fine plus it was working last week. I installed `juliaup` using the curl command and have been using for the past year without any issues. Any help would be much appreciated!

System
macos (intel) Ventura 13.4.1

Julia error message after repl temporarily appears
```julia
 ERROR: TaskFailedException
Stacktrace:
  [1] try_yieldto(undo::typeof(Base.ensure_rescheduled))
    @ Base ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/lib/julia/sys.dylib:-1
  [2] wait()
    @ Base ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/lib/julia/sys.dylib:-1
  [3] wait(c::Base.GenericCondition{ReentrantLock}; first::Bool)
    @ Base ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/lib/julia/sys.dylib:-1
  [4] take_buffered(c::Channel{Any})
    @ Base ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/lib/julia/sys.dylib:-1
  [5] repl_backend_loop(backend::REPL.REPLBackend, get_module::Function)
    @ REPL ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/lib/julia/sys.dylib:-1
  [6] start_repl_backend(backend::REPL.REPLBackend, consumer::Any; get_module::Function)
    @ REPL ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/lib/julia/sys.dylib:-1
  [7] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool, backend::Any)
    @ REPL ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/lib/julia/sys.dylib:-1
  [8] run_repl(repl::REPL.AbstractREPL, consumer::Any)
    @ REPL ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:365
  [9] (::Base.var"#1017#1019"{Bool, Bool, Bool})(REPL::Module)
    @ Base ./client.jl:421
 [10] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [11] invokelatest
    @ ./essentials.jl:813 [inlined]
 [12] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
    @ Base ./client.jl:405
 [13] exec_options(opts::Base.JLOptions)
    @ Base ./client.jl:322
 [14] _start()
    @ Base ./client.jl:522

    nested task error: Unhandled Task ERROR: IOError: stream is closed or unusable
Stacktrace:
 [1] check_open(x::Base.TTY)
   @ Base ./stream.jl:388
 [2] raw!(t::REPL.Terminals.TTYTerminal, raw::Bool)
   @ REPL.Terminals ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/Terminals.jl:138
 [3] prompt!(term::REPL.Terminals.TextTerminal, prompt::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
   @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2763
 [4] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
   @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2642
 [5] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
   @ REPL ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:1300
 [6] (::REPL.var"#62#68"{REPL.LineEditREPL, REPL.REPLBackendRef})()
   @ REPL ./task.jl:514

caused by: IOError: read: i/o error (EIO)
Stacktrace:
  [1] wait_readnb(x::Base.TTY, nb::Int64)
    @ Base ./stream.jl:410
  [2] eof(s::Base.TTY)
    @ Base ./stream.jl:106
  [3] eof(io::REPL.Terminals.TTYTerminal)
    @ Base ./io.jl:450
  [4] match_input(k::Dict{Char}, s::Union{Nothing, REPL.LineEdit.MIState}, term::Union{REPL.Terminals.AbstractTerminal, IOBuffer}, cs::Vector{Char}, keymap::Dict{Char})
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1619
  [5] match_input(k::Dict{Char}, s::Union{Nothing, REPL.LineEdit.MIState}, term::Union{REPL.Terminals.AbstractTerminal, IOBuffer}, cs::Vector{Char})
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1619
  [6] match_input(k::Dict{Char}, s::Union{Nothing, REPL.LineEdit.MIState}, term::Union{REPL.Terminals.AbstractTerminal, IOBuffer})
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1619
  [7] match_input(k::Dict{Char}, s::Union{Nothing, REPL.LineEdit.MIState})
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1619
  [8] prompt!(term::REPL.Terminals.TextTerminal, prompt::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2732
  [9] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2642
 [10] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:1300
 [11] (::REPL.var"#62#68"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ./task.jl:514
IOError: stream is closed or unusable
    Stacktrace:
     [1] check_open(x::Base.TTY)
       @ Base ./stream.jl:388
     [2] raw!(t::REPL.Terminals.TTYTerminal, raw::Bool)
       @ REPL.Terminals ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/Terminals.jl:138
     [3] prompt!(term::REPL.Terminals.TextTerminal, prompt::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
       @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2763
     [4] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
       @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2642
     [5] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
       @ REPL ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:1300
     [6] (::REPL.var"#62#68"{REPL.LineEditREPL, REPL.REPLBackendRef})()
       @ REPL ./task.jl:514
    
    caused by: IOError: read: i/o error (EIO)
    Stacktrace:
      [1] wait_readnb(x::Base.TTY, nb::Int64)
        @ Base ./stream.jl:410
      [2] eof(s::Base.TTY)
        @ Base ./stream.jl:106
      [3] eof(io::REPL.Terminals.TTYTerminal)
        @ Base ./io.jl:450
      [4] match_input(k::Dict{Char}, s::Union{Nothing, REPL.LineEdit.MIState}, term::Union{REPL.Terminals.AbstractTerminal, IOBuffer}, cs::Vector{Char}, keymap::Dict{Char})
        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1619
      [5] match_input(k::Dict{Char}, s::Union{Nothing, REPL.LineEdit.MIState}, term::Union{REPL.Terminals.AbstractTerminal, IOBuffer}, cs::Vector{Char})
        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1619
      [6] match_input(k::Dict{Char}, s::Union{Nothing, REPL.LineEdit.MIState}, term::Union{REPL.Terminals.AbstractTerminal, IOBuffer})
        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1619
      [7] match_input(k::Dict{Char}, s::Union{Nothing, REPL.LineEdit.MIState})
        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1619
      [8] prompt!(term::REPL.Terminals.TextTerminal, prompt::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2732
      [9] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2642
     [10] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
        @ REPL ~/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:1300
     [11] (::REPL.var"#62#68"{REPL.LineEditREPL, REPL.REPLBackendRef})()
        @ REPL ./task.jl:514
```

**Comment by jbisits** [comment]
By changing the path from the `julialauncher` set by `juliaup`
```termial
 path=('/Users/username/.julia/juliaup/bin' $path)
```
 to point to the binary of the version I wish to use
```terminal
 path=('/Users/myusername/.julia/juliaup/julia-1.9.2+0.x64.apple.darwin14/bin' $path)
```
I can now run julia, which solves my main problem. However I can still not execute `juliaup` commands and would like to use the `julialauncher` to access different versions. The error I previously saw when running `juliaup` commands persists eg
```terminal
juliaup list
zsh: permission denied: juliaup
```
which seems more related to some change in permissions that occurred when, or as a result, of the most recent version of `juliaup` version.

**Comment by DilumAluthge** [comment]
@StefanKarpinski @davidanthoff Likely related to our recent changes? Maybe `sysctl` is blocked if SIP is enabled?

**Comment by davidanthoff** [comment]
No, I think this is unrelated as @StefanKarpinski's edit should only affect new installs, but this seems like an existing install that should never touch the code that was changed.

I think the first order of business is to get the command `juliaup status` to work. I think as long as that doesn't work we won't make progress. The error messages that you get when trying to run `julialauncher` are probably knock-off problems that probably should be resolved once we get commands like `juliaup status` to work.

The thing that is confusing me is that you get 
```
juliaup list
zsh: permission denied: juliaup
```
Does that mean that the error message is actually from `zhs`? Do you get a similar error message if you run `juliaup status`?

What happens if you double click `juliaup` in finder?

**Comment by jbisits** [comment]
Thanks for the help! The error message running `juliaup status` is

```
 juliaup status
zsh: killed     juliaup status
```

Clicking on the `juliaup` executable in finder I get this message

```
 /Users/Joey/.julia/juliaup/bin/juliaup ; exit;
zsh: killed     /Users/Joey/.julia/juliaup/bin/juliaup

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
```

which as you mention looks to be related to `zsh`. The `zsh killed` message seems to be something that can [happen with homebrew](https://apple.stackexchange.com/questions/441000/most-homebrew-installed-commands-outputs-zsh-killed) and/or could be that something has become corrupted somewhere that is causing these issues for me.

I did try commenting out all lines in my `.zshrc` file and I get the same error

```
/Users/Joey/.julia/juliaup/bin/juliaup ; exit;
Joey@Josefs-Macbook-pro ~ % /Users/Joey/.julia/juliaup/bin/juliaup ; exit;
zsh: killed     /Users/Joey/.julia/juliaup/bin/juliaup

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
```

and this error for `julialauncher`

```
Joey@Josefs-Macbook-pro ~ % /Users/Joey/.julia/juliaup/bin/julialauncher ; exit;
Error: Failed to run version db update

Caused by:
    0: Failed to start juliaup for version db update.
    1: Malformed Mach-o file (os error 88)

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
```

The other thing I tried was using a bash shell but I got the same error running the `juliaup` executable from finder

```
Josefs-Macbook-pro:~ Joey$ /Users/Joey/.julia/juliaup/bin/juliaup ; exit;
Killed: 9
logout

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
```

There is some info about the [Killed: 9](https://apple.stackexchange.com/questions/258623/how-to-fix-killed-9-error-in-mac-os) error message that says
>If you replace a signed macOS binary by using cp instead of mv then macOS caches the signature, doesn't like the look of it because the file changed and kills your process when you try and start the new binary.

Could this have anything to do with the `juliaup` executable being moved from `.juliaup` into the home directory to `.julia/juliaup`?

Sorry for all the error messages! I will keep looking into why this might be happening at my end, but if you need any more info do let me know!

**Comment by jbisits** [comment]
@davidanthoff I solved this issue by removing the `.julia/juliaup` folde using finder in macos (I could not use any `juliaup` commands in the terminal so `juliaup self uninstall` could not be used), removing the `juliaup` commands from the `.zshrc` file then re-installing `juliaup` using `curl` (this was how I originally installed it). Everything now works as it should! The curios thing is that when `juliaup` was not working (as in when I filed this issue) there was no `.juliaup` file in my home directory as there is now after the new installation. Instead all `juliaup` related things were in the `.julia/juliaup` folder. Even though the `PATH` was correct in the `.zshrc`

```terminal
path=('/Users/Joey/.julia/juliaup/bin' $path)
export PATH
```

it would not work so perhaps it was a shell problem at my end that I could not figure out. 

But can confirm that after fresh install everything is working as it should so will close this issue.
